src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs

```diff
@@ -227,58 +159,45 @@ public override void OnStopActivity(Activity activity, object payload)
+                    activity.DisplayName = template;
+                    activity.SetTag(SemanticConventions.AttributeHttpRoute, template);
                 }
 
                 try
                 {
-                    this.options.Enrich?.Invoke(activityToEnrich, "OnStopActivity", response);
+                    this.options.Enrich?.Invoke(activity, "OnStopActivity", response);
                 }
                 catch (Exception ex)
                 {
-                    AspNetInstrumentationEventSource.Log.EnrichmentException(ex);
+                    AspNetInstrumentationEventSource.Log.EnrichmentException("OnStopActivity", ex);
                 }
             }
+        }
 
-            if (isCustomPropagator)
+        private void OnException(Activity activity, HttpContext context, Exception exception)
+        {
+            if (activity.IsAllDataRequested)
             {
-                if (activity.OperationName.Equals(ActivityNameByHttpInListener, StringComparison.Ordinal))
+                if (this.options.RecordException)
                 {
-                    // If instrumentation started a new Activity, it must
-                    // be stopped here.
-                    activity.Stop();
+                    activity.RecordException(exception);
+                }
 
-                    // Restore the original activity as Current.
-                    var activityByAspNet = (Activity)activity.GetCustomProperty("OTel.ActivityByAspNet");
-                    Activity.Current = activityByAspNet;
+                activity.SetStatus(Status.Error.WithDescription(exception.Message));
+
+                try
+                {
+                    this.options.Enrich?.Invoke(activity, "OnException", exception);
                 }
-                else if (createdActivity != null)
+                catch (Exception ex)
                 {
-                    // This block is hit if Asp.Net did restore Current to its own activity,
-                    // then we need to retrieve the one created by HttpInListener
-                    // and stop it.
-                    createdActivity.Stop();
-
-                    // Restore current back to the one created by Asp.Net
-                    Activity.Current = activity;
+                    AspNetInstrumentationEventSource.Log.EnrichmentException("OnException", ex);
                 }
             }
-        }
-
-        /// <summary>
-        /// Gets the OpenTelemetry standard uri tag value for a span based on its request <see cref="Uri"/>.
-        /// </summary>
-        /// <param name="uri"><see cref="Uri"/>.</param>
-        /// <returns>Span uri value.</returns>
-        private static string GetUriTagValueFromRequestUri(Uri uri)
-        {
-            if (string.IsNullOrEmpty(uri.UserInfo))
+            else
             {
-                return uri.ToString();
+                activity.SetStatus(Status.Error);
```

Comment:
@cijothomas @alanwest Check this out. In the case of exception thrown when  `!activity.IsAllDataRequested` I thought we should still at least mark the status as `Error`. What you guys think about that?

I'll do the same for AspNetCore if we all agree this is desired: https://github.com/open-telemetry/opentelemetry-dotnet/blob/0959e562d6af1fac5e178607b2ec85956ac5722c/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L267

Reply:
I ended up removing this. Turns out anything with `!activity.IsAllDataRequested` isn't fed to the processor(s) so it seemed kind of pointless to do.